bigc/resources/customers.py

```diff
@@ -81,6 +85,40 @@ def update_form_field(self, customer_id: int, field_name: str, value: Any) -> di
         }]
         return self._api.v3.put('/customers/form-field-values', json=payload)[0]
 
-    def delete(self, customer_id: int) -> None:
-        """Delete a specific customer by its ID"""
-        self._api.v3.delete(f'/customers?id:in={customer_id}')
+    def get_addresses(self, customer_id: int) -> dict:
+        """Get all addresses, filtering for a single customer"""
+        return self._api.v3.get(f'/customers/addresses?customer_id:in={customer_id}')
```

Comment:
* Following our naming conventions, this should be called `all_addresses`
* The return type should be `Iterator[dict]`
* You should be using `_api.v3.get_many()` to handle pagination